### PR TITLE
Block bound inventory keys from auto assignment

### PR DESCRIPTION
**Pull request by eJ0opb**
#### Summary

SUMMARY: Interface "Block bound inventory keys from auto assignment"

#### Purpose of change

It is very annoying when auto inventory letter assignment assigns over your preferred navigation keys.

#### Describe the solution

Check if the letter is bound to an inventory action before automatically assigning it.

#### Describe alternatives you've considered

* Turning off auto inventory letters
* Limiting myself to 7 favorites
* Buying a keyboard with arrow keys

#### Testing

1. Set auto inventory letters to enabled (favorites does not currently work in a way that I understand)
2. Create a character with more than 7 starting items
3. Observe that your navigation keys are not assigned
4. Manually assign your navigation keys.
5. Navigation keys select their assigned items instead of navigating

#### Additional context

Favorites and auto inventory letters seem to currently be in a bad state, so testing is limited.

Changing your key bindings after the letters have already been assigned doesn't work well.  I managed to finagle it into doing the right thing by manually unassigning everything, stripping naked, and picking everything back up again.

### CODE REVIEW

**Comment by Lamandus** [comment]
will that also happen if you don't have hjkl assigned? Maybe it should be linked to the buttons you have for movement. Some prefer even WASD

**Comment by ymber** [comment]
This should block any keys currently bound to menu navigation.

**Comment by enaantd** [comment]
> Maybe it should be linked to the buttons you have for movement. Some prefer even WASD

agreed, or even custom movement keys for those with an exotic keyboard layout (in my case I move with `u-p-e-x`). it would be a shame to hardcode this feature

**Comment by eJ0opb** [comment]
Who has familiarity with the inventory input code?  I looked at it, and I could not figure out how to find the relevant keybindings in a reasonable amount of time.

File: src/inventory.cpp [hunk 1]
```diff
@@ -1036,11 +1037,19 @@ void inventory::assign_empty_invlet( item &it, const Character &p, const bool fo
         }
     }
     if( cur_inv.count() < inv_chars.size() ) {
+        // XXX YUCK I don't know how else to get the keybindings
```
Comment (anothersimulacrum) [inline src/inventory.cpp:1040]: Fixme's are easy to search for.
```suggestion
        // XXX YUCK I don't know how else to get the keybindings
        // FIXME: Find a better way to get bound keys
```

File: src/inventory.cpp [hunk 2]
```diff
@@ -1036,11 +1037,20 @@ void inventory::assign_empty_invlet( item &it, const Character &p, const bool fo
...
+        // FIXME: Find a better way to get bound keys
+        avatar u;
```
Comment (KorGgenT) [inline src/inventory.cpp:1042]:
```suggestion
        avatar &u = g->u;
```
at the very least it should be the player character

**Pull request merged**